core/src/main/scala/kafka/common/RecordValidationException.scala

```diff
@@ -23,5 +23,6 @@ import org.apache.kafka.common.requests.ProduceResponse.RecordError
 import scala.collection.Seq
 
 class RecordValidationException(val invalidException: ApiException,
-                                val recordErrors: Seq[RecordError]) extends RuntimeException {
+                                val recordErrors: Seq[RecordError])
+  extends RuntimeException(invalidException) {
```

Comment:
I vaguely remember @rajinisivaram has another PR which may be overlapping with this fix, will let her to check and confirm if it is fine.

Reply:
Yeah, doesn't need to be done here. I just noticed the trace was missing when debugging a failure.

Reply:
I had only changed client-side code, so changes here wouldn't have conflicted.